### PR TITLE
zdoom: refactor

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Comment by aanderse** [comment]
@GrahamcOfBorg eval

**Review by aanderse — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108299` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>zdbsp</li>
<li>zdoom</li>
</ul>
</details>

File: pkgs/games/zdoom/zdbsp.nix [hunk 1]
```diff
@@ -21,7 +21,7 @@ stdenv.mkDerivation rec {
     description = "ZDoom's internal node builder for DOOM maps";
     homepage = "https://zdoom.org/wiki/ZDBSP";
     license = licenses.gpl2Plus;
-    maintainers = with maintainers; [ertes];
-    platforms = platforms.linux;
+    maintainers = with maintainers; [ ertes ];
```
Comment (Lassulus) [inline pkgs/games/zdoom/zdbsp.nix:24]: oh man, we need to remove ertes from maintainers someday

Comment (SuperSandro2000) [reply pkgs/games/zdoom/zdbsp.nix:24]: He was pretty active the last year.

Comment (Lassulus) [reply pkgs/games/zdoom/zdbsp.nix:24]: https://byorgey.wordpress.com/2018/05/21/ertugrul-soylemez-1985-2018/

Comment (siraben) [reply pkgs/games/zdoom/zdbsp.nix:24]: Sorry to hear that. I will take over maintenance of this package then.

Comment (siraben) [reply pkgs/games/zdoom/zdbsp.nix]: @Lassulus would you like to be added to `zdbsp` maintainers?

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/games/zdoom/zdbsp.nix (hunk 1 above)
Comment (Lassulus) [reply pkgs/games/zdoom/zdbsp.nix]: sure, why not

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108299` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>zdbsp</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108299` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>zdbsp</li>
<li>zdoom</li>
</ul>
</details>

**Pull request merged**